plugins/backstage-plugin-devcontainers-react/src/components/DevcontainersProvider/DevcontainersProvider.test.tsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { renderHook } from '@testing-library/react';
+import {
+  DEFAULT_DEVCONTAINERS_TAG,
+  DevcontainersConfig,
+  DevcontainersProvider,
+  useDevcontainersConfig,
+} from './DevcontainersProvider';
+
+const baseConfig: DevcontainersConfig = { tagName: 'test' };
+
+describe(`${DevcontainersProvider.name}`, () => {
+  it('Stabilizes the memory reference for the config value when defined outside the component', () => {
+    const { result, rerender } = renderHook(useDevcontainersConfig, {
+      wrapper: ({ children }) => (
+        <DevcontainersProvider config={baseConfig}>
+          {children}
+        </DevcontainersProvider>
+      ),
+    });
+
+    const initialResult = result.current;
+    rerender();
+    expect(result.current).toBe(initialResult);
+  });
+
+  it('Will update the memory reference for the config each render if it is accidentally passed inline', () => {
```

Comment:
This is not something we can actually control, is it?  Just React being React?  It feels like we are testing React rather than our plugin, if that makes sense.  But, not opposed to it if this would catch some kind of bug in our code!

Reply:
Yeah, it's one of those safety nets, where ideally the user would define the config outside the component, but I wanted to verify that things don't break if they forget about it. There's a similar test in the Coder plugin, too

Reply:
Yup I remember the matching Coder plugin test, but thinking about it more my intuition is that testing library code is fairly atypical, especially when testing its implementation details.  Otherwise, I start to think about where the line is drawn, which parts of React do we test and which do we trust?

Reply:
Yeah, maybe this is more of a philosophy thing, but I guess the way I see it is that the test case doesn't actually matter all that much, but it serves more as an enforceable form of documentation. 

As in, there's only so much that the type layer can communicate, so the test cases help communicate additional qualities and behaviors that the code should have

I wouldn't test this behavior for the main Coder app, but just because this is library code that's going to be imported into who-knows-what by people who may or may not know React all that well, I feel like it's a low-investment safety net that can help protect against freak regressions

Reply:
The documentation point is compelling, but it does seem weird to me that we should document React behavior, unless it is something obscure that we are using or relying on in the plugin code itself.

Though, I think maybe where I am getting hung up is that we specifically test whether the memory reference is the same.  Should we just check that it renders as expected in both cases?  Whether React is reusing the reference is not something we would care about, I think, we just want our plugin to show up properly regardless of how React chooses to do things internally.

Reply:
I do think the test could make more sense if React was not a peer dependency; since we only test a single version it will be difficult to catch a regression here I think.

Reply:
Yeah...you've convinced me
I guess my concern was that I wanted to make sure that we were accounting for as many possibilities as we could, no matter how someone chose to use our plugin. But I also feel like this might be something you just naturally have to vigilant about when working in a React codebase, so we're needless reinforcing things that plugin consumers should be doing anyway

Reply:
That does make sense!  I think I was just getting hung up on testing internals without seeing the bigger picture, if we keep the tests but test the visual result it seems most reasonable to me!  It might be unlikely to fail, and I suppose only consumers using our same React version would benefit, but honestly I struggle to trust React (after writing my nth accidental infinite loop) so making sure it is behaving does appeal to me.  :laughing: